pkg/revision/dependency.go

```diff
@@ -14,17 +14,33 @@
 
 package revision
 
-import v1 "github.com/istio-ecosystem/sail-operator/api/v1"
+import (
+	v1 "github.com/istio-ecosystem/sail-operator/api/v1"
+	"github.com/istio-ecosystem/sail-operator/pkg/config"
+)
+
+type computeValuesFunc func(*v1.Values, string, string, config.Platform, string, string, string, string) (*v1.Values, error)
+
+var defaultComputeValues computeValuesFunc = ComputeValues
 
 // DependsOnIstioCNI returns true if CNI is enabled in the revision
-func DependsOnIstioCNI(rev *v1.IstioRevision) bool {
-	// TODO: get actual final values and inspect pilot.cni.enabled
-	values := rev.Spec.Values
-	if values == nil {
+func DependsOnIstioCNI(rev *v1.IstioRevision, cfg config.ReconcilerConfig) bool {
+	values, err := defaultComputeValues(rev.Spec.Values, rev.Spec.Namespace, rev.Spec.Version,
+		cfg.Platform, cfg.DefaultProfile, "", cfg.ResourceDirectory, rev.Name)
+	if err != nil || values == nil {
```

Comment:
Is it a good idea to squash the error here?

It might be masking a bug..

I'd at least log this in case of error, if not cancel reconciliation attempt altogether.

WDYT?

Reply:
As this PR is fixing a blocker bug. Is it okay if I handle it in a follow up PR?

Reply:
@sridhargaddam I was going to say "sure" but I guess I have no choice in the matter..

Anyway might be a good idea to do something if it errors out, unless you think it makes sense to squash it (in which case maybe its good to add a comment why it's squashed).